src/interactive-window/interactiveWindow.ts

```diff
@@ -173,42 +177,48 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
             // This cannot happen, but we need to make typescript happy.
             throw new Error('Controller not selected');
         }
-        if (this.currentKernelInfo.kernel) {
-            return this.currentKernelInfo.kernel.promise;
+        if (this.currentKernelInfo.kernelStarted) {
+            return this.currentKernelInfo.kernelStarted.promise.then(() => this.currentKernelInfo.kernel!);
```

Comment:
```suggestion
            await this.currentKernelInfo.kernelStarted.promise
            return this.currentKernelInfo.kernel!;
```

This seems more intuitive to me.